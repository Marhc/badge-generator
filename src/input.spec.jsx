// @flow

import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import InputWithStyle from './input';
import { type Service, services, styles } from './config';

Enzyme.configure({ adapter: new Adapter() });

const noop = () => {};
const serviceSelection = services.map((service, index) => !!(index % 2));

describe('Input', () => {
  it('renders', () => {
    const component = renderer.create(
      <InputWithStyle
        repository="REPOSITORY"
        style={styles[0]}
        serviceSelection={serviceSelection}
        handleRepositoryChange={noop}
        handleServiceToggle={noop}
        handleStyleChange={noop}
      />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('calls handleRepositoryChange when respository changes', () => {
    const handleRepositoryChangeMock = jest.fn();
    const input = mount(
      <InputWithStyle
        repository="REPOSITORY"
        style={styles[0]}
        serviceSelection={serviceSelection}
        handleRepositoryChange={handleRepositoryChangeMock}
        handleServiceToggle={noop}
        handleStyleChange={noop}
      />,
    );

    expect(handleRepositoryChangeMock).toHaveBeenCalledTimes(0);
    input.find('input[type="text"]').simulate('change');
    expect(handleRepositoryChangeMock).toHaveBeenCalledTimes(1);
  });

  it('calls handleServiceToggle when a service is toggled', () => {
    const handleServiceToggleMocks = services.map(() => jest.fn());
    const input = mount(
      <InputWithStyle
        repository="REPOSITORY"
        style={styles[0]}
        serviceSelection={serviceSelection}
        handleRepositoryChange={noop}
        handleServiceToggle={index => handleServiceToggleMocks[index]}
        handleStyleChange={noop}
      />,
    );

    services.forEach((service: Service, index: number) => {
      expect(handleServiceToggleMocks[index]).toHaveBeenCalledTimes(0);
      input
        .find('input[type="checkbox"]')
        .at(index)
        .simulate('change');
      expect(handleServiceToggleMocks[index]).toHaveBeenCalledTimes(1);
    });
  });

  it('calls handleStyleChange when respository changes', () => {
    const handleStyleChangeMock = jest.fn();
    const input = mount(
      <InputWithStyle
        repository="REPOSITORY"
        style={styles[0]}
        serviceSelection={serviceSelection}
        handleRepositoryChange={noop}
        handleServiceToggle={noop}
        handleStyleChange={handleStyleChangeMock}
      />,
    );

    expect(handleStyleChangeMock).toHaveBeenCalledTimes(0);
    input
      .find('input[type="radio"][value="plastic"]')
      .simulate('change', { target: { checked: true } });
    expect(handleStyleChangeMock).toHaveBeenCalledTimes(1);
  });
});
