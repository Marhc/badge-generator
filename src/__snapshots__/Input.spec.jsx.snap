// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders its components 1`] = `
<form
  autoComplete="off"
  noValidate={true}
>
  <WithStyles(Grid)
    container={true}
  >
    <WithStyles(Grid)
      className=""
      item={true}
      xs={12}
    >
      <TextField
        fullWidth={true}
        helperText="Ex: \\"lodash\\" (npm name), \\"facebook/react\\" (Github slug), etc."
        label="Repository"
        margin="normal"
        onChange={[Function]}
        required={false}
        select={false}
        title="Repository"
        value="REPOSITORY"
        variant="standard"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className=""
      item={true}
      sm={10}
      xs={12}
    >
      <WithStyles(FormControl)
        component="fieldset"
      >
        <WithStyles(WithFormControlContext(FormLabel))
          component="legend"
        >
          Services
        </WithStyles(WithFormControlContext(FormLabel))>
        <WithStyles(Grid)
          container={true}
        >
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="0"
                />
              }
              label="GitHub forks"
              title="GitHub forks"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="1"
                />
              }
              label="GitHub stars"
              title="GitHub stars"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="2"
                />
              }
              label="GitHub license"
              title="GitHub license"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="3"
                />
              }
              label="Travis CI"
              title="Travis CI"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="4"
                />
              }
              label="CircleCI"
              title="CircleCI"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="5"
                />
              }
              label="CodeShip"
              title="CodeShip"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="6"
                />
              }
              label="Code Climate (maintainability)"
              title="Code Climate (maintainability)"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="7"
                />
              }
              label="Code Climate (% maintainability)"
              title="Code Climate (% maintainability)"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="8"
                />
              }
              label="Code Climate (tech debt)"
              title="Code Climate (tech debt)"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="9"
                />
              }
              label="Code Climate (code coverage)"
              title="Code Climate (code coverage)"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="10"
                />
              }
              label="Coveralls"
              title="Coveralls"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="11"
                />
              }
              label="Codecov"
              title="Codecov"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="12"
                />
              }
              label="Gemnasium"
              title="Gemnasium"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="13"
                />
              }
              label="VersionEye"
              title="VersionEye"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="14"
                />
              }
              label="David DM"
              title="David DM"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="15"
                />
              }
              label="David DM (dev)"
              title="David DM (dev)"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="16"
                />
              }
              label="David DM (peer)"
              title="David DM (peer)"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="17"
                />
              }
              label="Packagist"
              title="Packagist"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="18"
                />
              }
              label="Packagist PHP version"
              title="Packagist PHP version"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="19"
                />
              }
              label="Packagist License"
              title="Packagist License"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="20"
                />
              }
              label="NPM"
              title="NPM"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="21"
                />
              }
              label="NPM License"
              title="NPM License"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="22"
                />
              }
              label="NPM Total Downloads"
              title="NPM Total Downloads"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="23"
                />
              }
              label="NPM Downloads per month"
              title="NPM Downloads per month"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="24"
                />
              }
              label="Bower"
              title="Bower"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="25"
                />
              }
              label="ESDoc"
              title="ESDoc"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={false}
                  onChange={[Function]}
                  value="26"
                />
              }
              label="Greenkeeper"
              title="Greenkeeper"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={3}
            md={4}
            sm={6}
            xs={12}
          >
            <WithStyles(WithFormControlContext(FormControlLabel))
              control={
                <WithStyles(Checkbox)
                  checked={true}
                  onChange={[Function]}
                  value="27"
                />
              }
              label="Melpa"
              title="Melpa"
            />
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </WithStyles(FormControl)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className=""
      item={true}
      sm={2}
      xs={12}
    >
      <WithStyles(FormControl)
        component="fieldset"
      >
        <WithStyles(WithFormControlContext(FormLabel))
          component="legend"
        >
          Style
        </WithStyles(WithFormControlContext(FormLabel))>
        <RadioGroup
          aria-label="Style"
          name="style"
          onChange={[Function]}
          value="default"
        >
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={<WithStyles(Radio) />}
            label="Default"
            title="Default"
            value="default"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={<WithStyles(Radio) />}
            label="Plastic"
            title="Plastic"
            value="plastic"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={<WithStyles(Radio) />}
            label="Flat square"
            title="Flat square"
            value="flat-square"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={<WithStyles(Radio) />}
            label="For the badge"
            title="For the badge"
            value="for-the-badge"
          />
          <WithStyles(WithFormControlContext(FormControlLabel))
            control={<WithStyles(Radio) />}
            label="Social"
            title="Social"
            value="social"
          />
        </RadioGroup>
      </WithStyles(FormControl)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</form>
`;
